Meadow/Creatures/ScavengerController.cs

```diff
@@ -32,12 +32,12 @@ private static void ScavengerHand_Update(ILContext il)
                 for (int i = 0; i < 2; i++)
                 {
                     c.GotoNext(MoveType.After,
-                        i => i.MatchLdarg(0),
-                        i => i.MatchCall<ScavengerGraphics.ScavengerHand>("get_scavenger"),
-                        i => i.MatchLdfld<UpdatableAndDeletable>("room"),
-                        i => i.MatchLdloc(out _),
-                        i => i.MatchCallOrCallvirt<Room>("FloatWaterLevel")
-                        );
+                        i => i.MatchCallvirt<Creature>(typeof(Creature).GetProperty(nameof(Creature.mainBodyChunk)).GetGetMethod().Name),
```

Comment:
ILHooks accept MethodInfo
```suggestion
i => i.MatchCallvirt<Creature>(typeof(Creature).GetProperty(nameof(Creature.mainBodyChunk)).GetGetMethod()),
```

Reply:
It does not, throws compiler error

Reply:
It does you just have to remove the generic specification (I updated the my comment)